src/utils/read-repo-url.js

```diff
@@ -51,7 +50,10 @@ const parseRepoURL = function (repoHost, URL) {
   if (repoHost === GITHUB) {
     // https://developer.github.com/v3/repos/contents/#get-contents
     // what if it's not master? note that our contents retrieval may assume it is master
-    const [ownerAndRepo, contentsPath] = URL.path.split('/tree/master')
+    const [ownerAndRepo, contentsPath] = URL.path.split('/tree/master').map((path) =>
+      // remove the leading and trailing slashes
+      path.replace(/^\/?(.*?)\/?$/, '$1'),
+    )
```

Comment:
Since URL is expected to be in the format `https://github.com/netlify-labs/all-the-functions/tree/master/functions/9-using-middleware`, how about doing the following?

```suggestion
    const [ownerAndRepo, contentsPath] = URL.path.slice(1).split('/tree/master/')
```

Reply:
This is shorter and way more clearer - thanks!